zrml/prediction-markets/src/lib.rs

```diff
@@ -1845,29 +1845,22 @@ mod pallet {
             let one_read = T::DbWeight::get().reads(1);
             let one_write = T::DbWeight::get().writes(1);
 
-            let retain_closure =
-                |subsidy_info: &SubsidyUntil<T::BlockNumber, MomentOf<T>, MarketIdOf<T>>| {
-                    let market_ready = match &subsidy_info.period {
-                        MarketPeriod::Block(period) => period.start <= current_block,
-                        MarketPeriod::Timestamp(period) => period.start <= current_time,
-                    };
+            let retain_closure = |subsidy_info: &SubsidyUntil<
+                T::BlockNumber,
+                MomentOf<T>,
+                MarketIdOf<T>,
+            >| {
+                let market_ready = match &subsidy_info.period {
+                    MarketPeriod::Block(period) => period.start <= current_block,
+                    MarketPeriod::Timestamp(period) => period.start <= current_time,
+                };
 
-                    if market_ready {
-                        let pool_id = T::MarketCommons::market_pool(&subsidy_info.market_id);
-                        total_weight.saturating_add(one_read);
+                if market_ready {
+                    let pool_id = T::MarketCommons::market_pool(&subsidy_info.market_id);
+                    total_weight.saturating_add(one_read);
 
-                        if let Err(err) = pool_id {
-                            log::error!(
-                                "[PredictionMarkets] Cannot find pool associated to market.
-                            market_id: {:?}, error: {:?}",
-                                pool_id,
-                                err
-                            );
-                            return true;
-                        }
-
-                        let end_subsidy_result =
-                            T::Swaps::end_subsidy_phase(pool_id.unwrap_or(u128::MAX));
+                    if let Ok(pool_id) = pool_id {
```

Comment:
Feels better than the mitigation suggestion, because if someone would have changed this in the future, we can not be sure, that the `expect` panic would have been caught (although, the tests would fail). It's more code capsulation here now. I am thinking about reducing the code capsulation with functions, but maybe this refactoring is for another PR. 

> What you could do is shadow the variable right after the if-condition a la let pool_id = pool_id.expect("early return happens above in case of Err");. Then you can remove the two .unwrap_or(u128::MAX) in the function.

Reply:
I probably like the `match` here more (because it's even more clear, that there are no more cases than `Ok()` or `Err()`), but this would lead to even more code capsulation, so you can leave it as it is.